### ISSUE TITLE
Problem with downloading subtitles from NBC

### ISSUE THREAD

**Issue by m0ck69** [yt-dlp/yt-dlp#7478]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.06.22** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

USA

### Provide a description that is worded well enough to be understood

There is a problem with downloading subtitles from NBC.com
Sometimes yt-dlp returns `Requested format is not available` although the subtitles are available on the site.
Example link:` https://www.nbc.com/los-herederos-del-monte/video/obligados-a-separarse/4193804`

Second issue. Sometimes there are two subtitles available (English and Spanish) but yt-dlp downloads only one.
Example link:` https://www.nbc.com/donde-esta-elisa/video/identifican-al-secuestrador/3714109`
I;m not sure but I think NBC may mark both subtitle tracks as "eng" although one is not English. Is there a way to workaround this?

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://www.nbc.com/los-herederos-del-monte/video/obligados-a-separarse/4193804', '--all-subs', '--convert-subs', 'srt', '--skip-download', '--sub-langs', 'all']
[debug] Encodings: locale cp1250, fs utf-8, pref cp1250, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.06.22 [812cdfa06] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg git-2019-11-22-27c6c92
[debug] Optional libraries: Cryptodome-3.18.0, brotli-1.0.9, certifi-2023.05.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-11.0.3
[debug] Proxy map: {}
[debug] Loaded 1851 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.06.22, Current version: stable@2023.06.22
Current Build Hash: 37e7ffe204309357cfd1388b0e2c782a30e293ddd0f2761a9a8f6afa185b3566
yt-dlp is up to date (stable@2023.06.22)
[NBC] Extracting URL: https://www.nbc.com/los-herederos-del-monte/video/obligados-a-separarse/4193804
[NBC] 4193804: Downloading JSON metadata
[NBC] 4193804: Downloading JSON metadata
[ThePlatform] Extracting URL: http://link.theplatform.com/s/NnzsPC/media/guid/2304991196/4193804?mbr=true&manifest=m3u&switch=HLSServiceSecure#__youtubedl_smuggle=%7B%22force_smil_url%22%3A+true%7D
[ThePlatform] 4193804: Downloading SMIL data
[ThePlatform] 4193804: Downloading m3u8 information
WARNING: [ThePlatform] Ignoring subtitle tracks found in the HLS manifest; if any subtitle tracks are missing, please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[ThePlatform] 4193804: Downloading JSON metadata
[info] 4193804: Downloading subtitles: en
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
ERROR: [ThePlatform] 4193804: Requested format is not available. Use --list-formats for a list of available formats
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1555, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1652, in __extract_info
  File "yt_dlp\YoutubeDL.py", line 1764, in process_ie_result
  File "yt_dlp\YoutubeDL.py", line 1711, in process_ie_result
  File "yt_dlp\YoutubeDL.py", line 2783, in process_video_result
yt_dlp.utils.ExtractorError: [ThePlatform] 4193804: Requested format is not available. Use --list-formats for a list of available formats
```

**Comment by bashonly** [comment]
> Sometimes yt-dlp returns Requested format is not available although the subtitles are available on the site.

this is because you are requesting `eng` when the language code of the available subs is `en`

> Sometimes there are two subtitles available (English and Spanish) but yt-dlp downloads only one.

this is a valid issue and it's likely due to ThePlatform extractor's use of `_extract_smil_formats`, which only calls `_extract_m3u8_formats` instead of `_extract_m3u8_formats_and_subtitles`

> WARNING: [ThePlatform] Ignoring subtitle tracks found in the HLS manifest; if any subtitle tracks are missing,

**Comment by qzmtch** [comment]
> `best` is supposed to have a special behavior when all formats are video-only. I'll take a look at that.(but not the actual extractor issues)

hello, now I don’t have the opportunity and time to issue a completely separate report, so I’ll write here
it turns out for a long time there is a bug in yt-dlp, with a choice of format
so there is no error `yt-dlp -f "bestvideo[height<=1080][ext=mp4]+bestaudio[ext=m4a]"`
and here is the error `yt-dlp -f "bestvideo[height=1080][ext=mp4]+bestaudio[ext=m4a]"`
if you remove < and leave only = then there will be an error 
"ERROR: [youtube] DBod1MLTDKc: Requested format is not available. Use --list-formats for a list of available formats"

**Comment by pukkandan** [comment]
> `best` is supposed to have a special behavior when all formats are video-only. I'll take a look at that

Fixed in https://github.com/yt-dlp/yt-dlp/commit/906c0bdcd8974340d619e99ccd613c163eb0d0c2

**Comment by bashonly** [comment]
Closed by 550e65410a7a1b105923494ac44460a4dc1a15d9
